### PR TITLE
Add serveIndex option

### PR DESCRIPTION
**Pull request by jantimon**
Adds the feature request #11 with tests.

Moved default options to module.exports.getDefaultOptions.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -49,25 +58,45 @@ module.exports = function(opts) {
...
   assert(opts.baseDir, 'opts.baseDir should be provided');
 
-  var parsed = url.parse(req.originalUrl);
+  var parsed = url.parse(requestUrl);
   var pathname = parsed.pathname.replace(opts.baseUrl, '');
 
-  for (var i = 0; i < opts.ext.length; i++) {
-    var ext = opts.ext[i];
-    var filepath = path.join(opts.baseDir, pathname.replace(/\.html$/, ext));
+  if (opts.serveIndex && pathname.substr(-1) === '/') {
```
Comment (runk) [inline index.js]: what if someone request

`/views`

instead of

`/views/`
?

File: index.js [hunk 2]
```diff
@@ -49,25 +58,45 @@ module.exports = function(opts) {
...
+    pathname += 'index' + opts.allowedExt[0];
+  }
+
+  var requestedExt = path.extname(pathname);
+  var requestedFilename = pathname.substr(0, pathname.length - requestedExt.length);
```
Comment (runk) [inline index.js]:
```
requestedFilename = path.basename(pathname, requestedExt);
```

Should do a trick as well http://nodejs.org/api/path.html#path_path_basename_p_ext

**Comment by runk** [comment]
Looking awesome, just a couple of improvements/questions. Love the tests! :+1:

File: index.js (hunk 1 above)
Comment (jantimon) [inline index.js]: Fixed this in the latest commit

File: index.js (hunk 2 above)
Comment (jantimon) [inline index.js]: It would however I need also the path. I changed the misleading variable name

**Comment by runk** [comment]
Great, merging :+1:

**Comment by runk** [comment]
published v0.2

**Pull request merged**